macro/src/lib.rs

```diff
@@ -375,20 +414,27 @@ impl FieldParser {
             pub fn #pointer_fn_name(&mut self) -> *mut #type_path{
                 unsafe { core::ptr::addr_of_mut!((*self.ptr).#field_ident) }
             }
-
-            #[doc = "Read the "]
-            #[doc = concat!("[", stringify!(#ident), "::", stringify!(#field_ident), "]")]
-            #[doc = " register."]
-            #[inline(always)]
-            pub fn #read_fn_name(&mut self) -> #type_path {
-                let addr = self.#pointer_fn_name();
-                unsafe {
-                    addr.read_volatile()
-                }
-            }
         });
+        if let Some(read_access) = access.read {
+            let mut opt_mut = TokenStream::new();
+            if read_access == ReadAccess::Normal {
+                opt_mut.append_all(quote! { mut });
+            }
```

Comment:
I think you can do the following
```suggestion
        if let Some(read_access) = access.read {
            let opt_mut = (read_access == ReadAccess::Normal).then_some(syn::Token![mut]);
```
and then in the `quote!` using it
```diff
-                pub fn #read_fn_name(&#opt_mut self) -> #type_path {
+                pub fn #read_fn_name(&#(#opt_mut)? self) -> #type_path {
```

(same for the equivalent changes around line 500)

Reply:
Ah right, its `#(..)*` still, `?` isn't a thing for quote sorry. `quote` can treat options as iterators so that should work out as expected

Reply:
if it is an option, #(...) also works. I have to use `quote! { mut }` though. I have tried `syn::Token[mut]` in combination with quote and `#(...)`  / #(...)`